projects/plugins/jetpack/_inc/blogging-prompts.php

```diff
@@ -47,6 +47,33 @@ function jetpack_setup_blogging_prompt_response( $post_id ) {
 
 add_action( 'wp_insert_post', 'jetpack_setup_blogging_prompt_response' );
 
+/**
+ * Add a meta value to posts when publishd that answer a blogging prompt.
+ *
+ * @param string  $new_status New post status.
+ * @param string  $old_status Old post status.
+ * @param WP_Post $post       Post object.
+ */
+function jetpack_mark_if_post_answers_blogging_prompt( $new_status, $old_status, $post ) {
+	// Make sure we are publishing a post, and it's not already published.
+	if ( 'post' !== $post->post_type || 'publish' !== $new_status || 'publish' === $old_status ) {
+		return;
+	}
+
+	$blocks = parse_blocks( $post->post_content );
+	foreach ( $blocks as $block ) {
+		if ( 'jetpack/blogging-prompt' === $block['blockName'] ) {
+			$prompt_id = $block['attrs']['promptId'];
```

Comment:
this logs a warning if `$block['attrs']['promptId']` is undefined, I think you should do an `isset()` check.